First role: ci_make

cjeanner opened the pull request:
This role purpose is to generate indexed playbooks running "make" commands in a specific location.

That way, we will be able to reproduce the various "make" runs in a convenient, ordered way.

raukadah commented on ci_framework/roles/ci_make/molecule/default/prepare.yml:
```diff
@@ -0,0 +1,31 @@
+        path: /tmp/foo
+        state: directory
+    - name: Push dummy Makefile
+      ansible.builtin.file:
+        target: /tmp/foo/Makefile
+        content: |
+          FOO_BAR ?= "toto"
+          help:
+            echo "This is the help thing showing ${FOO_BAR}"
```
- name: push dummy makefile
  ansible.builtin.copy:
      dest:  /tmp/foo/Makefile
       content: |
          FOO_BAR ?= "toto"
          help:
            echo "This is the help thing showing ${FOO_BAR}"

raukadah commented on ci_framework/roles/ci_make/molecule/default/molecule.yml:
```diff
@@ -0,0 +1,28 @@
+---
+driver:
+  name: podman
+
```
We need to add platforms here:
`platforms:
  - name: centos
    hostname: centos
    pre_build_image: true
    image: ${TRIPLEO_ANSIBLE_MOLECULE_IMAGE:-"ubi9/ubi-init"}
    registry:
      url: ${TRIPLEO_ANSIBLE_MOLECULE_REGISTRY:-"registry.access.redhat.com"}`

raukadah commented on ci_framework/roles/ci_make/defaults/main.yml:
```diff
@@ -0,0 +1,20 @@
+
+# All variables intended for modification should be placed in this file.
+# All variables within this role should have a prefix of "cifmw_ci_make"
+cifmw_ci_make_outputdir: "{{ ansible_user_directory }}/ci_output"
```
s/ansible_user_directory/ansible_user_dir

cjeanner replied:
oh. damn me.

raukadah commented on ci_framework/roles/ci_make/molecule/default/converge.yml:
```diff
@@ -0,0 +1,30 @@
+    - name: Default role with chdir
+      vars:
+        chdir: "/tmp/foo"
+      include_role: ci_make
```
include_role:
   name: ci_make

cjeanner replied:
actually I had an issue yesterday with my local branch, and didn't see it didn't push ^^'. Sorry!

raukadah commented on ci_framework/roles/ci_make/molecule/default/converge.yml:
```diff
@@ -0,0 +1,30 @@
+        chdir: "/tmp/foo"
+        params:
+          FOO_BAR: "starwars"
+      include_role: ci_make
```
```
include_role:
   name: ci_make
```

raukadah reviewed (commented):
It is a great start.
Since it is still in draft phase, I will stop adding more comment.

cjeanner commented:
/recheck

cjeanner commented:
Running this: ```make run_ctx_molecule BUILD_VENV_CTX=no MOLECULE_CONFIG=.config/molecule/config_local.yml```

ends with:
```Bash
[...]
PLAY [Prepare] *****************************************************************

TASK [Create lame directory] ***************************************************
changed: [instance]

TASK [Push dummy Makefile] *****************************************************
changed: [instance]

PLAY RECAP *********************************************************************
instance                   : ok=2    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

INFO     Running default > converge

PLAY [Converge] ****************************************************************

TASK [Gathering Facts] *********************************************************
ok: [instance]

TASK [Default role with chdir] *************************************************

TASK [ci_make : Validate we get at least chdir param] **************************
skipping: [instance]

TASK [ci_make : Ensure output directory exists] ********************************
changed: [instance]

TASK [ci_make : Get current amount of playbooks] *******************************
ok: [instance]

TASK [ci_make : Create filename fact] ******************************************
ok: [instance]

TASK [ci_make : Create new make call using index and target] *******************
changed: [instance]

TASK [ci_make : Call make module if wanted] ************************************
changed: [instance]

TASK [ci_make : Expose make output] ********************************************
ok: [instance]

TASK [Validate make output] ****************************************************
ok: [instance] => changed=false 
  msg: All assertions passed

TASK [Role with specific parameter] ********************************************

TASK [ci_make : Validate we get at least chdir param] **************************
skipping: [instance]

TASK [ci_make : Ensure output directory exists] ********************************
ok: [instance]

TASK [ci_make : Get current amount of playbooks] *******************************
ok: [instance]

TASK [ci_make : Create filename fact] ******************************************
ok: [instance]

TASK [ci_make : Create new make call using index and target] *******************
changed: [instance]

TASK [ci_make : Call make module if wanted] ************************************
changed: [instance]

TASK [ci_make : Expose make output] ********************************************
ok: [instance]

TASK [Validate make output] ****************************************************
ok: [instance] => changed=false 
  msg: All assertions passed

PLAY RECAP *********************************************************************
instance                   : ok=15   changed=5    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0
```


I guess this is what one may call a great success :).

raukadah commented:
/lgtm
